### ISSUE TITLE
[FreeBSD] salt-call on 2019.2.0+ does not return shell prompt if using Python3

### ISSUE THREAD

**Issue by cedwards** [saltstack/salt#56332]
### Description of Issue
After migrating to py37 on FreeBSD `salt-call` behaves strangely in that it does not return the shell prompt after the successful (or failed) application of the state. This does not affect py27 installations.

### Steps to Reproduce Issue
`pkg install py37-salt` on FreeBSD on use `salt-call state.apply foo`.

### Versions Report
```
Salt Version:
           Salt: 2019.2.2

Dependency Versions:
           cffi: 1.13.2
       cherrypy: Not Installed
       dateutil: 2.8.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.7.6 (default, Jan 30 2020, 01:17:40)
   python-gnupg: Not Installed
         PyYAML: 5.2
          PyZMQ: 18.1.1
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1

System Versions:
           dist:
         locale: UTF-8
        machine: amd64
        release: 11.3-RELEASE
         system: FreeBSD
        version: Not Installed

```

**Comment by Ch3LL** [comment]
are you also seeing this on 3000?

**Comment by cedwards** [comment]
Have not looked at 3000.

**Comment by krionbsd** [comment]
I can reproduce it locally as well with:
```
Salt Version:                                                                                                                           
           Salt: 2019.2.3                                                                                                               
                                                                                                                                        
Dependency Versions:                                                                                                                    
           cffi: 1.14.0                                                                                                                 
       cherrypy: Not Installed                                                                                                          
       dateutil: Not Installed                                                                                                          
      docker-py: Not Installed                                                                                                          
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.7.6 (default, Jan 13 2020, 14:10:17)
   python-gnupg: Not Installed
         PyYAML: 5.2
          PyZMQ: 18.1.1
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist:   
         locale: utf-8
        machine: amd64
        release: 13.0-CURRENT
         system: FreeBSD
        version: Not Installed
```

**Comment by Ch3LL** [comment]
and to clarify this only affects salt-call? it does not affect any other salt commands?

**Comment by cedwards** [comment]
So far I have only seen the issue with `salt-call` on the minion and not with `salt` from the master.

**Comment by krionbsd** [comment]
and I've to add `salt-call` is affected only while running `state.apply state`, running `salt-call` + execution modules works as expected

**Comment by Ch3LL** [comment]
thanks for the clarifications

**Comment by twangboy** [comment]
@cedwards What is the state you're running? Some commands cause `communicate` to fail. See my comments on a possibly related issue: https://github.com/saltstack/salt/issues/56231#issuecomment-614901232

**Comment by ari** [comment]
This happens for me on every state and also highstate. FreeBSD 12.1. Salt 2019.2.5

**Comment by krionbsd** [comment]
I can't reproduce this failure on salt3000.3 which I'm going to port to FreeBSD in the next days. 2019.2.5 is still affected and has this issue though, stay tuned we're working on it.

**Comment by krionbsd** [comment]
As we moved to 3000.3 in `sysutils/py-salt` this issue is fixed there. @cedwards any plans to migrate to 3000.3?

**Comment by sagetherage** [comment]
closing this issue as fixed in 3000.3
